Specialization/Unholy.lua

```diff
@@ -132,51 +121,25 @@ function DeathKnight:Unholy()
 	fd.timeTo4Runes = timeTo4Runes;
 	fd.deathCoilRP = deathCoilRP;
 
-	DeathKnight:UholyGlowCooldowns();
-
-	-- outbreak,if=dot.virulent_plague.refreshable&!talent.unholy_blight&!raid_event.adds.exists;
-	if runes >= 1 and (debuff[UH.VirulentPlague].refreshable and not talents[UH.UnholyBlight] and targets <= 1) then
-		return UH.Outbreak;
-	end
-
-	-- outbreak,if=dot.virulent_plague.refreshable&active_enemies>=2&(!talent.unholy_blight|talent.unholy_blight&cooldown.unholy_blight.remains);
-	if runes >= 1 and
-		debuff[UH.VirulentPlague].refreshable and
-		targets >= 2 and
-		(not talents[UH.UnholyBlight] or (talents[UH.UnholyBlight] and not cooldown[UH.UnholyBlight].ready))
-	then
-		return UH.Outbreak;
-	end
-
-	-- outbreak,if=runeforge.superstrain&(dot.frost_fever.refreshable|dot.blood_plague.refreshable);
-	if runes >= 1 and
-		runeforge[UH.SuperstrainBonusId] and
-		(debuff[UH.FrostFever].refreshable or debuff[UH.BloodPlague].refreshable)
-	then
-		return UH.Outbreak;
-	end
-
-	-- NOTE: This isn't in the simc priority list, but several people noted that Outbreak was never used if they had Unholy Blight as a talent, even when UB was on a long cooldown.
-	if runes >= 1 and
-		debuff[UH.VirulentPlague].remains < gcd and
-		talents[UH.UnholyBlight] and
-		(cooldown[UH.UnholyBlight].remains > 10 or (talents[UH.VirulentPlague] and cooldown[UH.UnholyBlight].remains > 8))
-	then
-		return UH.Outbreak;
+	DeathKnight:UnolyGlowCooldowns();
+	
+	if not fd.petExists then
+		return UH.RaiseDeadUnholy;
 	end
-
-	-- actions+=/wait_for_cooldown,name=soul_reaper,if=talent.soul_reaper&target.time_to_pct_35<5&fight_remains>5&cooldown.soul_reaper.remains<(gcd*0.75)&active_enemies=1
-	-- NOTE: It appears that wait_for_cooldown is designed to perform the action as soon as it's off cooldown.  We can't do that exactly the same way, but we can prevent
-	--       highlighting any abilities, and since Soul Reaper has a high priority, the end result should be the same.
-	-- TODO: Implement time-to-percent logic
-	-- TODO: Implement fight-remains logic
-	if talents[UH.SoulReaper] and
-		targetHpPercent <= 35 and
-		cooldown[UH.SoulReaper].remains < (gcd * 0.75) and
-		not cooldown[UH.SoulReaper].ready and
-		targets == 1
-	then
-		return;
+	
+	--outbreak or unholy blight
+	if fd.debuff[UH.VirulentPlague].refreshable and fd.runes > 0 then
+		if fd.targets < 2 then
+			return UH.Outbreak;
+		end
+		if fd.targets > 1 then
```

Comment:
General comment: I would suggest either using local variables, or always referring the framedata, but not a mixture. It makes the code longer.

Reply:
I agree. I'll update with local variables since that is already the established pattern.